content/running_bookwyrm/cli.md

```diff
@@ -0,0 +1,143 @@
+Title: Command Line Tool
+Date: 2021-11-06
+Order: 6
+
+Bookwyrm developers and instance managers can use the `bw-dev` script for common tasks. This can make your commands shorter, easier to remember, and harder mess up.
+
+Once you have installed Bookwyrm [in production](installing-in-production.html) or [in development](https://docs.joinbookwyrm.com/developer-environment.html#setting_up_the_developer_environment), you can run the script from the command line with `./bw-dev` followed by the subcommand you want to run.
+
+## Docker shortcuts
+
+### bash
+
+Open an interactive `bash` session inside the docker `web` container.
+
+### build
+
+Equivalent to `docker-compose build`.
+
+### dbshell
+
+Open an interactive Postgres database shell. I hope you know what you're doing.
+
+### run
+
+Run an arbitrary command in the `web` cointainer.
+
+Equivalent to `docker-compose run --rm --service-ports web`. 
+
+### rundb
+
+TODO: this command does not appear to actually exist!
+
+### runweb _args_
+
+Run an arbitrary command in the `web` container.
+
+Equivalent to `docker-compose run --rm web`.
+
+### shell
+
+Open an interactive Django shell inside the docker `web` container. You would use this if you want to run Django shell commands directly. For example when [making your initial user a superuser](installing-in-production.html#configure_your_instance).
+
+### up [args]
+
+Start or restart Docker containers. Equivalent to `docker-compose up --build [args]`
+
+## Managing the database
+
+### initdb
+
+Initialize a database.
+
+### makemigrations
+
+Runs Django's `makemigrations` command inside your Docker container. If you have changed the database structure in a development branch you will need to run this for your changes to have effect.
+
+### migrate
+
+Runs Django's `migrate` command inside your Docker container. You always need to run this after `makemigrations`.
+
+### resetdb
```

Comment:
maybe this one should go under the development and testing header? It shouldn't be available on the production branch

Reply:
Hmm now I'm thinking in most cases all the database management commands are really for dev and test? If you're managing an instance you'd probably just use `update`?

Reply:
Ideally you'd just use `update`, but in practice, they can be useful in production when you run into a situation like a merge conflict with local changes that `update` can't anticipate. I think it makes sense to have a section separate from the other development commands, but I don't feel strongly about it :)

Reply:
I've left this in but put a note about it only being available in prod.